includes/database/QMySqliDatabase.class.php

```diff
@@ -1,4 +1,8 @@
 <?php
+
+if (!defined('MYSQLI_ON_UPDATE_NOW_FLAG')) {
+	define ('MYSQLI_ON_UPDATE_NOW_FLAG', 8192);
```

Comment:
Is this really safe? If php (or the mysql driver) really added then deleted this flag, maybe this value is reserved for something else in the future. There is basically no way for us to know if the `8192` value does what we think it does.
Isn't it better to do something like this around the lines where the value is used:
<pre>
if (defined(MYSQLI_ON_UPDATE_NOW_FLAG)) {
	// Special situation that we take advantage of to automatically implement optimistic locking
	if ($mixFieldData->type == MYSQLI_TYPE_TIMESTAMP &&
			$mixFieldData->flags & MYSQLI_ON_UPDATE_NOW_FLAG) {
		$this->strType = QDatabaseFieldType::VarChar;
		$this->blnTimestamp = true;
	}
}
</pre>